AstaZero/src/OverlappingReferenceLines.cpp

```diff
@@ -0,0 +1,86 @@
+#include "OverlappingReferenceLines.hpp"
+
+OverlappingReferenceLines::OverlappingReferenceLines(const char* openDriveFileName, double deltaS)
+	: CalculateReferenceLine(openDriveFileName, deltaS) {
+	double error;
+	std::string output_file_name = "test2.csv";
+	std::ofstream file;
+	file.open(output_file_name);
+	std::string sampling_step = "1.0";
+	static char strbuf[1024];
+	for (auto it = openDriveReferenceLines.begin(); it != openDriveReferenceLines.end(); ++it) {
+		if (std::next(it, 1) == openDriveReferenceLines.end())
+			break;
+		error = rootMeanSquareError(it->second, (std::next(it, 1)->second));
+		if (error < 1 && compareGeometryHeading(it->first, std::next(it, 1)->first, 170.0, 190.0)) {
+			file << "lane, " << it->first->GetId() << ", " << 0 << ", " << 0 << ", driving" << std::endl;
+
+			std::cout << "Error Dist: " << error << " Road id1 :" << it->first->GetId()
+					  << " Road id2: " << std::next(it, 1)->first->GetId() << std::endl;
+			for (auto v : it->second) {
+				snprintf(strbuf, sizeof(strbuf), "%f, %f, %f, %f\n", v->x, v->y, 0, 0);
+				file << strbuf;
+			}
+			file << "lane, " << std::next(it, 1)->first->GetId() << ", " << 0 << ", " << 0 << ", driving"
+				 << std::endl;
+
+			for (auto v : std::next(it, 1)->second) {
+				snprintf(strbuf, sizeof(strbuf), "%f, %f, %f, %f\n", v->x, v->y, 0, 0);
+				file << strbuf;
+			}
+			overlappingRoads.insert(
+				std::pair<std::shared_ptr<Road>, std::shared_ptr<Road>>(it->first, std::next(it, 1)->first));
+		}
+	}
+}
+
+bool OverlappingReferenceLines::compareGeometryHeading(std::shared_ptr<Road> road1,
+													   std::shared_ptr<Road> road2,
+													   double min,
+													   double max) {
+	auto geometry1 = road1->getGeometryVector();
+	auto geometry2 = road2->getGeometryVector();
+	double differenceHeading0 = abs(geometry1[0]->GetHdg() - geometry2[0]->GetHdg());
+	double differenceHeadingEnd = abs(geometry1.back()->GetHdg() - geometry2.back()->GetHdg());
+	return (differenceHeading0 > min * M_PI / 180 && differenceHeading0 < max * M_PI / 180
+			&& differenceHeadingEnd > min * M_PI / 180 && differenceHeadingEnd < max * M_PI / 180);
```

Comment:
Why bother with the conversion to radians here four times? Either pass in radians or use degrees.